rest-api/test/unit_test/dao_test/questionnaire_response_dao_test.py

```diff
@@ -244,6 +259,39 @@ def test_insert_email_only(self):
     with self.assertRaises(BadRequest):
       self.questionnaire_response_dao.insert(qr)
 
+  def test_insert_login_phone_number_only(self):
+    self.insert_codes()
+    p = Participant(participantId=1, biobankId=2)
+    self.participant_dao.insert(p)
+    self._setup_questionnaire()
+    qr = QuestionnaireResponse(questionnaireResponseId=1, questionnaireId=1, questionnaireVersion=1,
+                               participantId=1, resource=QUESTIONNAIRE_RESPONSE_RESOURCE)
+    qr.answers.append(self.LOGIN_PHONE_NUMBER_ANSWER)
+    # First and last name are required.
+    with self.assertRaises(BadRequest):
+      self.questionnaire_response_dao.insert(qr)
+
+  def test_insert_both_names_and_login_phone_number(self):
```

Comment:
A couple more tests we could add:

- Both email and login phone number but missing first or last name (fails)
- Email and login phone number and first and last name (succeeds)